### PR TITLE
Plot Rydberg Densities & Calculate from `AnalogHamiltonianSimulationTaskResult`

### PR DESCRIPTION
**Pull request by johnzl-777**
Added `plot_densities` function that plots the Rydberg densities for each site over the atom positions. Also added new definition for `rydberg_density` to calculate densities from a Braket.jl `AnalogHamiltonianTaskResult`.

### CODE REVIEW

File: src/Bloqade.jl [hunk 1]
```diff
@@ -50,6 +51,9 @@ function __init__()
     return
 end
 
+using Colors, ColorSchemes
+using Braket: AnalogHamiltonianSimulationQuantumTaskResult
```
[the remark this answers is not in the document]
Comment (johnzl-777) [reply src/Bloqade.jl:55]: I figured it was 🐟 😅 , I'll give putting it into an extension a shot!

**Comment by johnzl-777** [comment]
Instead of the Package extensions and after talking more with @weinbe58 I've opted to do the following to minimize the number of new dependencies Bloqade has to handle and disturbing the existing API.

Considering `BloqadeSchema` is the main library for interacting with quantum hardware through `Bloqade` AND it already depends on `Braket.jl`:
* `rydberg_density` for `AnalogHamiltonianSimulationQuantumTaskResult` now lives in `BloqadeSchema` (it is not exported to avoid conflict with `rydberg_density` in base `Bloqade`)
* `plot_densities` has been generalized (it will just accept a vector of Rydberg densities versus having two definitions for two separate types), removing the direct dependence on `Braket.jl` and living in `plots.jl`.

The only additional cost is that Bloqade is now dependent on `Colors.jl` and `ColorSchemes.jl`.

File: test/plots.jl [hunk 2]
```diff
@@ -11,3 +11,19 @@ bitstring_hist(r; nlargest = 10)
 space = Subspace(5, randperm(1 << 5)[1:10] .- 1)
 r = rand_state(space)
 bitstring_hist(r; nlargest = 10)
+
+@testset "plot_densities" begin
```
Comment (weinbe58) [inline test/plots.jl:15]: can we also add a test that doesn't throw an error to make sure we cover that branch?

**Review by weinbe58 — CHANGES_REQUESTED** [review]

**Review by weinbe58 — APPROVED** [review]
LGTM

**Pull request merged**